include/pybind11/numpy.h

```diff
@@ -129,6 +129,11 @@ struct npy_api {
         NPY_STRING_, NPY_UNICODE_, NPY_VOID_
     };
 
+    typedef struct {
+        const Py_intptr_t *ptr;
```

Comment:
Just change the `const ShapeContainer &new_shape` to `ShapeContainer new_shape`; it's essentially always going to be passed by value (via implicit conversion), so the const reference doesn't gain anything.